src/renderers/shared/ReactDebugTool.js

```diff
@@ -283,90 +344,90 @@ var ReactDebugTool = {
     resumeCurrentLifeCycleTimer();
     emitEvent('onEndFlush');
   },
-  onBeginLifeCycleTimer(debugID, timerType) {
+  onBeginLifeCycleTimer(debugID: DebugID, timerType: TimerType): void {
     checkDebugID(debugID);
     emitEvent('onBeginLifeCycleTimer', debugID, timerType);
     markBegin(debugID, timerType);
     beginLifeCycleTimer(debugID, timerType);
   },
-  onEndLifeCycleTimer(debugID, timerType) {
+  onEndLifeCycleTimer(debugID: DebugID, timerType: TimerType): void {
     checkDebugID(debugID);
     endLifeCycleTimer(debugID, timerType);
     markEnd(debugID, timerType);
     emitEvent('onEndLifeCycleTimer', debugID, timerType);
   },
-  onBeginProcessingChildContext() {
+  onBeginProcessingChildContext(): void {
     emitEvent('onBeginProcessingChildContext');
   },
-  onEndProcessingChildContext() {
+  onEndProcessingChildContext(): void {
     emitEvent('onEndProcessingChildContext');
   },
-  onHostOperation(debugID, type, payload) {
+  onHostOperation(debugID: DebugID, type: HostOperationType, payload: Payload): void {
     checkDebugID(debugID);
     emitEvent('onHostOperation', debugID, type, payload);
   },
-  onSetState() {
+  onSetState(): void {
     emitEvent('onSetState');
   },
-  onSetChildren(debugID, childDebugIDs) {
+  onSetChildren(debugID: DebugID, childDebugIDs: Array<DebugID>) {
     checkDebugID(debugID);
     childDebugIDs.forEach(checkDebugID);
     emitEvent('onSetChildren', debugID, childDebugIDs);
   },
-  onBeforeMountComponent(debugID, element, parentDebugID) {
+  onBeforeMountComponent(debugID: DebugID, element: ReactElement, parentDebugID: DebugID): void {
     checkDebugID(debugID);
     checkDebugID(parentDebugID, true);
     emitEvent('onBeforeMountComponent', debugID, element, parentDebugID);
     markBegin(debugID, 'mount');
   },
-  onMountComponent(debugID) {
+  onMountComponent(debugID: DebugID): void {
     checkDebugID(debugID);
     markEnd(debugID, 'mount');
     emitEvent('onMountComponent', debugID);
   },
-  onBeforeUpdateComponent(debugID, element) {
+  onBeforeUpdateComponent(debugID: DebugID, element: ReactElement): void {
     checkDebugID(debugID);
     emitEvent('onBeforeUpdateComponent', debugID, element);
     markBegin(debugID, 'update');
   },
-  onUpdateComponent(debugID) {
+  onUpdateComponent(debugID: DebugID): void {
     checkDebugID(debugID);
     markEnd(debugID, 'update');
     emitEvent('onUpdateComponent', debugID);
   },
-  onBeforeUnmountComponent(debugID) {
+  onBeforeUnmountComponent(debugID: DebugID): void {
     checkDebugID(debugID);
     emitEvent('onBeforeUnmountComponent', debugID);
     markBegin(debugID, 'unmount');
   },
-  onUnmountComponent(debugID) {
+  onUnmountComponent(debugID: DebugID): void {
     checkDebugID(debugID);
     markEnd(debugID, 'unmount');
     emitEvent('onUnmountComponent', debugID);
   },
-  onTestEvent() {
+  onTestEvent(): void {
     emitEvent('onTestEvent');
   },
 };
 
 // TODO remove these when RN/www gets updated
```

Comment:
If we kill them, people who use old RN with new React will have issues. Unfortunately it seems like a lot of people still do that so I'd keep until a major.